wsmatrix@martin.zurowietz.de/ThumbnailsBoxOverride.js

```diff
@@ -71,37 +86,37 @@ var ThumbnailsBoxOverride = class {
    _activateThumbnailAtPoint(stageX, stageY, time) {
       let [r, x, y] = this.transform_stage_point(stageX, stageY);
 
-      for (let i = 0; i < this._thumbnails.length; i++) {
-         let thumbnail = this._thumbnails[i]
-         let [w, h] = thumbnail.actor.get_transformed_size();
-
-         // Add x range check to check for both vertical and horizontal position
-         if (y >= thumbnail.actor.y && y <= thumbnail.actor.y + h &&
-            x >= thumbnail.actor.x && x <= thumbnail.actor.x + w) {
-            thumbnail.activate(time);
-            break;
-         }
-      }
-      this.queue_relayout();
+      let thumbnail = this._thumbnails.find(t => {
+         let [w, h] = t.get_transformed_size();
+         return y >= t.y && y <= t.y + h && x >= t.x && x <= t.x + w;
+      });
+      if (thumbnail)
+         thumbnail.activate(time);
    }
 
    // Overriding the Tweener animation to consider both vertical and horizontal changes
    // the original method only animates vertically
    _activeWorkspaceChanged(wm, from, to, direction) {
-      let thumbnail;
       let workspaceManager = global.workspace_manager;
       let activeWorkspace = workspaceManager.get_active_workspace();
-      for (let i = 0; i < this._thumbnails.length; i++) {
-         if (this._thumbnails[i].metaWorkspace == activeWorkspace) {
-            thumbnail = this._thumbnails[i];
-            break;
-         }
-      }
+      let thumbnail = this._thumbnails.find(t => t.metaWorkspace == activeWorkspace);
+      let [thumbX, thumbY] = thumbnail.get_position();
 
       this._animatingIndicator = true;
-      this.indicatorY = thumbnail.actor.allocation.y1;
-      this.indicatorX = thumbnail.actor.allocation.x1;
-      this._animatingIndicator = false;
+
+      //todo: this code should animate x & y, but for some reason it is not
```

Comment:
Is this issue resolved? If we can't get the animation working, the code should be removed again.

Reply:
it's not really an issue, I am assuming this code should animate the thumbnail indicator (thumbnail border) in the overview. However, it's moving to the correct position without being animated.

I tried to debug but had no luck.